### DISCUSSION TITLE
附件选择选项卡扩展的图片不支持在图库中插入

### DISCUSSION THREAD

**Discussion by liuyiwuqing** [halo-dev/halo#6067] [Q&A]
### 系统信息

- 外部访问地址: https://blog.muyin.site
- 启动时间: 2024-06-09 21:49
- 版本: 2.16.3
- 构建时间: 2024-06-07 11:55
- Git Commit: 77f888d
- Java: IBM Semeru Runtime Open Edition / 21.0.3+9-LTS
- 数据库: MySQL Community Server - GPL / 8.2.0
- 操作系统: Linux / 6.1.0-13-amd64
- 已激活主题: [Earth 1.8.0](https://github.com/halo-dev/theme-earth.git)
- 已启动插件:
  - [编辑器超链接卡片 1.0.0](https://github.com/halo-sigs/plugin-editor-hyperlink-card)
  - [图床插件 1.0.2-SNAPSHOT1](https://github.com/liuyiwuqing/halo-plugin-picture-bed)
  - [highlight.js 代码高亮 1.2.1](https://github.com/halo-sigs/plugin-highlightjs)
  - [AI 助手 1.0.0](https://www.halo.run/store/apps/app-riNgb)
  - [静态网页服务 1.0.0-alpha.2](https://www.halo.run/store/apps/app-gFkMn)
  - [lightgallery.js 灯箱 1.2.0](https://github.com/halo-sigs/plugin-lightgallery)
  - [Vditor 编辑器 1.7.0](https://github.com/justice2001/halo-plugin-vditor)
  - [Tools工具箱 1.2.3-beta1](https://github.com/liuyiwuqing/halo-plugin-tools.git)
  - [文章限制阅读 1.2.4](https://www.halo.run/store/apps/app-IzWXo)
  - [TinyTale小程序设置 1.0.4](https://github.com/jiewenhuang/halo-plugin-TinyTale)
  - [爱发电插件 1.6.1](https://github.com/carolcoral/plugin-afdian)
  - [Lsky Pro 兰空图床 0.2.1](https://github.com/ichenhe/halo-lsky-pro)
  - [站点迁移 1.4.0](https://github.com/halo-sigs/plugin-migrate)
  - [Chirpy Add-on 0.0.2](https://github.com/airbozh/halo-plugin-chirpy)
  - [对象存储（Amazon S3 协议） 1.9.1](https://github.com/halo-dev/plugin-s3)
  - [SEO 工具集 1.0.5](https://www.halo.run/store/apps/app-FNGbT)
  - [增强备份（Backup IMproved） 1.1.1](https://www.halo.run/store/apps/app-dHakX)
  - [插件授权管理 1.0.5](https://github.com/liuyiwuqing/lywq-plugins-auth)
  - [友链自助提交插件 1.4.2](https://github.com/liuyiwuqing/halo-plugin-links-submit)
  - [站点推送插件 1.5.1](https://github.com/Stonewuu/halo-plugin-sitepush)
  - [Nps管理 1.3.1](https://github.com/liuyiwuqing/halo-plugin-nps)
  - [Unsplash 1.1.0](https://github.com/halo-sigs/plugin-unsplash)
  - [图库管理 1.3.0](https://github.com/halo-sigs/plugin-photos)
  - [Docsme 1.0.0-alpha.6](https://www.halo.run/store/apps/app-yffxw)
  - [Markdown / HTML 内容块 1.2.0](https://github.com/halo-dev/plugin-hybrid-edit-block)
  - [朋友圈 1.3.2](https://github.com/chengzhongxue/plugin-friends)
  - [Metrics Graph 1.0.0](https://github.com/guqing/plugin-metrics-graph)
  - [链接管理 1.3.3](https://github.com/halo-sigs/plugin-links)
  - [OAuth2 认证 1.1.1](https://github.com/halo-sigs/plugin-oauth2)
  - [瞬间 1.6.1](https://github.com/halo-sigs/plugin-moments)
  - [Sitemap 1.1.1](https://github.com/halo-dev/plugin-sitemap)
  - [搜索组件 1.4.0](https://github.com/halo-dev/plugin-search-widget)
  - [RSS 1.2.2](https://github.com/halo-dev/plugin-feed)
  - [评论组件 2.2.0](https://github.com/halo-dev/plugin-comment-widget)
  - [应用市场 1.2.0](https://www.halo.run/store/apps/app-VYJbF)

### 使用的哪种方式运行？

Docker Compose

### 发生了什么？

根据插件开发文档，对附件选择选项卡进行拓展实现的时候，发现在图库中无法插入图片，提示`只支持选择图片`，
参考文档：[附件选择选项卡](https://docs.halo.run/developer-guide/plugin/api-reference/ui/extension-points/attachment-selector-create)
插件复现：[halo-plugin-picture-bed](https://github.com/liuyiwuqing/halo-plugin-picture-bed/issues/2)
类似插件：[plugin-unsplash](https://github.com/halo-sigs/plugin-unsplash)

**Comment by ruibaby** [comment]
我这边使用你的插件发现 sm.ms 是可以的，但是 issue 中提到的兰空图床图床我没有环境进行测试，你可以先看看图库插件对此的判断：https://github.com/halo-sigs/plugin-photos/blob/36aafb1b0e79e888f2f94e5c994e162677f05dcb/console/src/views/PhotoList.vue#L261-L279

**Comment by ruibaby** [comment]
由于此 issue 不属于 Halo 本身的问题，我将转移到 Disscussion。
